python/lsst/faro/measurement/DetectorTableMeasurement.py

```diff
@@ -0,0 +1,129 @@
+    columns = pexConfig.Field(doc="Columns from sourceTable_visit to load.",
+                              dtype=str, default='coord_ra, coord_dec, detector')
+    refDataset = pexConfig.Field(doc="Reference dataset to use.",
+                                 dtype=str, default='')
+
+
+class DetectorTableMeasurementTask(CatalogMeasurementBaseTask):
+    ConfigClass = DetectorTableMeasurementTaskConfig
+    _DefaultName = "detectorTableMeasurementTask"
+
+    def run(self, catalog, dataIds, refcat, refcatCalib):
+        return self.measure.run(catalog, self.config.connections.metric, dataIds)
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+        columns = [_.strip() for _ in self.config.columns.split(',')]
+        catalog = inputs['catalog'].get(parameters={'columns': columns})
+        selection = (catalog['detector'] == butlerQC.quantum.dataId['detector'])
+        inputs['catalog'] = catalog[selection]
+        inputs['dataIds'] = [butlerQC.registry.expandDataId(inputRefs.catalog.datasetRef.dataId)]
+
+        if self.config.connections.refDataset != '':
+            config = LoadReferenceCatalogConfig()
+            config.refObjLoader.ref_dataset_name = self.config.connections.refDataset
+
+            # These default configurations for reference catalogs should go
+            # elsewhere in a common utility.
+            if self.config.connections.refDataset == 'gaia_dr2_20200414':
+                # Apply proper motions for Gaia catalog
+                config.refObjLoader.requireProperMotion = True
+                config.refObjLoader.anyFilterMapsToThis = 'phot_g_mean'
+                config.doApplyColorTerms = False
+            elif self.config.connections.refDataset == 'ps1_pv3_3pi_20170110':
+                # Apply color terms for PS1 catalog
+                config.refObjLoader.load(os.path.join(getPackageDir('obs_subaru'),
+                                                      'config',
+                                                      'filterMap.py'))
+                config.colorterms.load(os.path.join(getPackageDir('obs_subaru'),
+                                                    'config',
+                                                    'colorterms.py'))
+
+            # Directly concatenate the reference catalogs
+            # This approach does NOT match rows with LoadReferenceCatalogTask
+            # refCat = inputs['refcat'][0].get()
+            # for handle in inputs['refcat'][1:]:
+            #     refCat.extend(handle.get())
+            # if not refCat.isContiguous():
+            #     refCat = refCat.copy(deep=True)
+            # inputs['refcat'] = refCat
+
+            center = lsst.geom.SpherePoint(butlerQC.quantum.dataId.region.getBoundingCircle().getCenter())
+            radius = butlerQC.quantum.dataId.region.getBoundingCircle().getOpeningAngle()
+
+            epoch = butlerQC.quantum.dataId.records['visit'].timespan.begin
```

Comment:
Is this an astropy.date?

Reply:
`type(epoch)` returns `<class 'astropy.time.core.Time'>`